### PR TITLE
feat: add app initialization

### PR DESCRIPTION
**Pull request by dakotalillie**
**Describe your changes**

- Redirect user to `/init` if application is not initialized. Initialization is verified by checking whether there's an admin user. This check should happen on every page.
- To initialize the app, the user fills out the information for the admin user. Currently this does not include a password, because we do not yet support password authentication. My intention is to iterate on this and add that in a subsequent PR, as this one is already fairly large.
- If initialization is verified, cache the initialization status in a cookie to avoid making an extra database query on every page load.
- disable `react/jsx-no-bind` rule, which recommends that _every_ callback in a function component be wrapped with `useCallback`, which is [an antipattern](https://kentcdodds.com/blog/usememo-and-usecallback)
- disable `unicorn/prevent-abbreviations` for "prop" and "props", since nobody calls React props "properties"
- add a volume to `docker-compose.yml` to persist data between container restarts

**Testing performed**

- Only manual testing currently. If this approach is approved, I'll add the tests and convert this from a draft PR to a regular one.

**Additional context**

N/A

### CODE REVIEW

**Comment by dakotalillie** [comment]
Will look into fixing the build check failure tomorrow (I believe I just need to add some stubbed env values into the workflow config)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -85,6 +85,14 @@ jobs:
...
       - run: npm ci
       - run: npm run build
+        env:
+          OPENID_CLIENT_ID: 'client-id'
+          OPENID_CLIENT_SECRET: 'secret'
+          OPENID_ISSUER: 'issuer'
+          JWT_AUTHORITY_ISSUER: 'issuer'
+          JWT_AUTHORITY_ISSUER_SHARED_SECRET: 'secret'
+          SESSION_SECRET: 'secret'
+          NEXT_PUBLIC_GQL_BACKEND_URL: 'http://localhost:4444/graphql'
```
Comment (portenez) [inline .github/workflows/ci.yml:95]: Why do we need to use `NEXT_PUBLIC` but not in the other cases?

File: schema.graphql [hunk 2]
```diff
@@ -170,6 +170,7 @@ scalar MAC
 
 type Mutation {
   addUser(email: String!, given_name: String!, family_name: String!): User
+  initialize(email: String!, given_name: String!, family_name: String!): User
```
Comment (portenez) [inline schema.graphql:173]: Adding context per our discussion. During initialization, the user should claim themselves as admin. 

For the secondary admin. Let's do it in another PR.

File: src/pages/init.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  context
+) => {
+  const { req, res } = context
+
+  if (req.cookies[INITIALIZED_COOKIE_NAME]) {
+    return { props: { isInitialized: true } }
+  }
+
+  const { isInitialized } = await request<InitPageQuery>(
+    NEXT_PUBLIC_GQL_BACKEND_URL,
```
Comment (portenez) [inline src/pages/init.tsx:28]: Is `NEXT_PUBLIC_GQL_BACKEND_URL` only used from within `GetServerSideProps`? if so, should we change it back to `GQL_BACKEND_URL`?

I don't remember exactly what was it, but in the past, I've had issues with `NEXT_PUBLIC_*`. Maybe things have changed?

What's the benefit of prefixing with `NEXT_PUBLIC`, if it's not used on the browser side?

**Review by portenez — COMMENTED** [review]
A few comments here and there. 
 
 We also need to change to use the "self-claim" approach.

**Comment by dakotalillie** [comment]
Much has changed, will take another stab
